core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -777,6 +778,42 @@ private[spark] object Utils extends Logging {
   }
 
   /**
+   * This method compresses the files passed in, and writes the compressed data out into the
+   * [[OutputStream]] passed in. Each file is written as a new [[ZipEntry]] with its name being
+   * the name of the file being compressed.
+   */
+  private[spark] def zipFilesToStream(
```

Comment:
I'll double-check how this method is being used, but it needs at least some exception-safety to close the input streams it's creating.